src/plugin/manager.ts

```
@@ -0,0 +1,630 @@
+  ExtensionPointType,
+} from './types';
+import { validatePluginManifest, sortDrawerItems } from './utils';
+import { PluginGraphQLService, IPlugin } from './graphql-service';
+
+class PluginManager {
+  private loadedPlugins: Map<string, ILoadedPlugin> = new Map();
+  private extensionRegistry: IExtensionRegistry = {
+    routes: [],
+    drawer: [],
+  };
+  private eventListeners: Map<string, Function[]> = new Map();
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)